crates/ty/src/lib.rs

```diff
@@ -314,14 +319,16 @@ impl MainLoop {
                             let mut max_severity = Severity::Info;
                             let diagnostics_count = result.len();
 
+                            let mut stdout = self.printer.stdout().lock();
+
                             for diagnostic in result {
```

Comment:
Rendering diagnostics is fairly expensive (especially when using the `full` format). We should avoid iterating and rendering all diagnostics if we end up discarding the output anyway